src/averageAges.js

```diff
@@ -1,61 +1,64 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+// function to find only men including century specifying
+
+function filterMen(people, sex, century) {
+  return people.filter(person => {
+    return person.sex === sex && century
+      ? century === Math.ceil(person.died / 100)
+      : person.sex === sex;
+  });
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = filterMen(people, 'm', century);
+
+  return calculateAverageAges(filteredMen);
+}
+
+// function to find only women including children specifying
+
+function filterWomen(people, sex, withChildren) {
+  return people.filter(person => {
+    return person.sex === sex && withChildren
+      ? people.some(
+        ({ mother }) => mother === person.name
+      )
+      : person.sex === sex;
+  });
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = filterWomen(people, 'f', withChildren);
+
+  return calculateAverageAges(filteredWomen);
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(
+    person => people.some(
+      mother => {
+        return mother.name === person.mother && onlyWithSon
+          ? person.sex === 'm'
+          : mother.name === person.mother;
```

Comment:
Same here: `mother.name === person.mother` repeated twice. Make it a constant. And create `isWithSon` too.